module/zfs/gzip.c

```diff
@@ -49,13 +52,33 @@ typedef uLongf zlen_t;
 
 #endif
 
+#ifdef HAVE_QAT
+#define	QAT_MIN_BUF_SIZE	4096
+int
+use_qat(size_t s_len)
+{
```

Comment:
It would be nice to have a run time disable for this as a module option.  Say `zfs_qat_disable=0` by default.  See `zfs_prefetch_disable` as an example.